assets/src/hooks/useNotificationsReducer.ts

```diff
@@ -183,9 +195,82 @@ const persistToggledNotificationReadState = (
   ])
 }
 
-export const useNotificationsReducer = (): [State, Dispatch] => {
+type InitialNotificationData = { initial_notifications: NotificationData[] }
+
+type InitialNotifications = { type: "initial"; payload: Notification[] }
+type NewNotification = { type: "new"; payload: Notification }
+
+type ReceivedNotifications = NewNotification | InitialNotifications | null
+
+const parseNotifications = (
+  notificationData: NotificationData | InitialNotificationData
+): NewNotification | InitialNotifications => {
+  if ("initial_notifications" in notificationData) {
+    return {
+      type: "initial",
+      payload: (
+        notificationData as InitialNotificationData
+      ).initial_notifications.map(notificationFromData),
+    }
+  }
+  return {
+    type: "new",
+    payload: notificationFromData(notificationData as NotificationData),
+  }
+}
+
+const useNotifications = (socket: Socket | undefined, routeIds: RouteId[]) => {
```

Comment:
I'm thinking about breaking this out into a separate file to simplify `useNotificationsReducer` & independently testing it

Reply:
That would make sense (`useNotifications` lives to see another day! My original changes deleted it entirely).